### ISSUE TITLE
rrdesi in simulating spectra tutorial broken

### ISSUE THREAD

**Issue by sbailey** [desihub/redrock#61]
As reported by @paulmartini, the rrdesi step of https://github.com/desihub/tutorials/blob/master/Intro_to_DESI_spectra.ipynb runs redrock and then fails while collecting stuff up to write:
```
Traceback (most recent call last):
  File "/global/common/cori/contrib/desi/desiconda/current/code/redrock/master/bin/rrdesi", line 7, in <module>
    desi.rrdesi()
  File "/global/common/cori/contrib/desi/desiconda/current/code/redrock/master/py/redrock/external/desi.py", line 270, in rrdesi
    zbest['BRICKNAME'] = fibermap['BRICKNAME'][ii[0]]
  File "/global/common/cori/contrib/desi/desiconda/20170613-1.1.4-spectro/code/desiconda/20170613-1.1.4-spectro_conda/lib/python3.5/site-packages/astropy/table/table.py", line 1281, in __setitem__
    unit=getattr(value, 'unit', None))
  File "/global/common/cori/contrib/desi/desiconda/20170613-1.1.4-spectro/code/desiconda/20170613-1.1.4-spectro_conda/lib/python3.5/site-packages/astropy/table/column.py", line 707, in __new__
    copy=copy, copy_indices=copy_indices)
  File "/global/common/cori/contrib/desi/desiconda/20170613-1.1.4-spectro/code/desiconda/20170613-1.1.4-spectro_conda/lib/python3.5/site-packages/astropy/table/column.py", line 111, in __new__
    self_data = np.zeros(length, dtype=dtype)
ValueError: invalid itemsize in generic type tuple
srun: error: nid00182: task 0: Exited with exit code 1
srun: Terminating job step 9483287.0
In [13]:
```

To do:
1. Fix this.
2. Include a unit or integration test that would have caught it without running the tutorial.  Our current unit tests + spectro pipeline integration tests + end-to-end mini test do not catch this.

**Comment by sbailey** [comment]
This bug has already been fixed in the refactor branch.  In master (and 0.7.0) the problem only occurs when the input spectra don't have BRICKNAMEs.  There is some oddity with creating a new column of an astropy table when the new value is a blank string of numpy.bytes_.  It works with non-blank strings of numpy.bytes_ and works with blank strings of vanilla python bytes, and I don't want to wade into debugging astropy.

I'll leave this ticket open until verified after refactor branch is merged.  Having advocated for a unit test to catch this, it is such a irritating corner case that I'm not sure it is worth creating a test...

**Comment by sbailey** [comment]
fixed by PR #67
